tumblr/kanvas-ios
# Mute button control left of editing controls

Adds a mute button to the left of the editing controls.

![IMG_4422](https://user-images.githubusercontent.com/3250/108468518-16c0aa80-7244-11eb-98a8-2b402ac70c5b.jpg)

## Overview

* A new `EditorViewController.Edit` type with `NSSecureCoding` support replaces the old `Edit` type which held archived data. This type now includes new information about the muted state of the audio for saving and restoring locally and for exporting.
* A new Mute button control is added to the left of the normal controls container.
* A new `muteButton` Camera Setting determines whether to enable this setting (off by default).

## Testing

* Enable the "Mute button" setting
* Add a video
* Ensure that the mute button is shown
* Ensure that toggling the mute button starts and stops the playback of audio during preview
* Ensure that setting the enabling the mute button removes audio from the exported file
